qark/plugins/external_storage.py

```diff
@@ -0,0 +1,104 @@
+import sys
+import os
+import re
+sys.path.insert(0, os.path.dirname(os.path.realpath(__file__)) + '../lib')
+from yapsy.IPlugin import IPlugin
+from plugins import PluginUtil
+from modules import common
+from lib.pubsub import pub
+import lib.plyj.parser as plyj
+
+
+class ExternalStorageCheckPlugin(IPlugin):
+
+    CHECK_EXTERNAL_STORAGE = r'getExternalFilesDir'
+    CHECK_EXTERNAL_MEDIA = r'getExternalMediaDirs'
+    CHECK_PUBLIC_DIR = r'getExternalStoragePublicDirectory'
+
+    def __init__(self):
+        self.name = 'External Storage Issues'
+
+    def target(self, queue):
+        files = common.java_files
+        global parser, tree, file_name
+        parser = plyj.Parser()
+        tree = ''
+        res = []
+        external_storage = []
+        external_media = []
+        external_pub_dir = []
+        count = 0
+        for file in files:
+            count += 1
+            pub.sendMessage('progress', bar=self.name, percent=round(count * 100 / len(files)))
+            file_name = str(file)
+            try:
+                tree = parser.parse_file(file)
+            except Exception:
+                continue
+            try:
+                for import_decl in tree.import_declarations:
+                    if 'File' in import_decl.name.value:
+                        with open(file_name, 'r') as f:
+                            file_body = f.read()
+                        if PluginUtil.contains(self.CHECK_EXTERNAL_STORAGE, file_body):
+                            external_storage.append(file_name)
+                            break
+
+                        if PluginUtil.contains(self.CHECK_EXTERNAL_MEDIA, file_body):
+                            external_media.append(file_name)
+                            break
+
+                        if PluginUtil.contains(self.CHECK_PUBLIC_DIR, file_body):
+                            external_pub_dir.append(file_name)
+                            break
+
+            except Exception:
+                continue
```

Comment:
Once you decide which Exceptions are really throw away versus ones that should be handled, you can `pass` on the specific ones and not the rest.